src/Development/IDE/Core/Shake.hs

```diff
@@ -439,27 +503,6 @@ getHiddenDiagnostics IdeState{shakeExtras = ShakeExtras{hiddenDiagnostics}} = do
     val <- readVar hiddenDiagnostics
     return $ getAllDiagnostics val
 
--- | FIXME: This function is temporary! Only required because the files of interest doesn't work
-unsafeClearDiagnostics :: IdeState -> IO ()
-unsafeClearDiagnostics IdeState{shakeExtras = ShakeExtras{diagnostics}} =
-    writeVar diagnostics mempty
-
--- | Clear the results for all files that do not match the given predicate.
-garbageCollect :: (NormalizedFilePath -> Bool) -> Action ()
```

Comment:
This is useful and should stay - it relates to how kick should work.

Reply:
It's not used anywhere though. Is the idea to garbage collect diagnostics for files that are no longer in the transitive closure of interest? Maybe the right thing to do then is to open a ticket to implement such behaviour (or at least to discuss whether that's the right thing to do).

Reply:
https://github.com/digital-asset/ghcide/issues/123 already tracks the issue. It's not got much detail to it yet though. We definitely want this feature, as its absence has bitten me in the last few days.

Reply:
I've opened https://github.com/digital-asset/ghcide/issues/580

Reply:
Please keep this and related functions for now, we still use this in `daml`.